opm/output/eclipse/WriteInit.cpp

```diff
@@ -673,24 +908,89 @@ void Opm::InitIO::write(const ::Opm::EclipseState&              es,
         if (es.globalFieldProps().has_double("MULTPV")) {
             multipliers.insert("MULTPV", UnitSystem::measure::identity,
                         es.globalFieldProps().get_double("MULTPV"),
-                        data::TargetType::INIT);
+                        data::TargetType::INIT); 
         }
 
         writeSimulatorProperties(grid, multipliers, initFile);
     }
 
-    writeTableData(es, units, initFile);
+    // ITEM 3 - LOOP FOR LOCAL PROPERTIES LGR
+    if (grid.is_lgr()) 
+    {
+        std::vector<std::string> all_lgr_tag  = grid.get_all_lgr_labels();
+        for (std::size_t index : grid.get_print_order_lgr()) 
+        {
+            auto lgr_label = all_lgr_tag[index];
+            const EclipseGridLGR& lgr_grid = grid.getLGRCell(lgr_label);
+            const std::array<int,3> subdivisions = grid.getCellSubdivisionRatioLGR(lgr_label);
+            std::vector<int> global_fathers = lgr_grid.getLGRCell_global_father(grid);
+            writeInitFileHeaderLGRCell(es, lgr_grid, schedule, initFile, index+1);           
+            writePoreVolumeLGRCell(es, units, initFile, global_fathers, subdivisions[0]*subdivisions[1]*subdivisions[2]);
+            writeGridGeometryLGRCell(grid, lgr_grid, units, initFile, subdivisions[0], subdivisions[1], subdivisions[2]);
+            writeDoubleCellProperties(es, units, initFile, global_fathers);
+            writeSimulatorPropertiesLGRCell(grid, simProps, initFile, global_fathers);
+            {
+                const auto writeAll = es.cfg().io().writeAllTransMultipliers();
+        
+                auto multipliers = es.getTransMult()
+                    .convertToSimProps(grid.getNumActive(), writeAll);
+                if (es.globalFieldProps().has_double("MULTPV")) {
+                    multipliers.insert("MULTPV", UnitSystem::measure::identity,
+                                es.globalFieldProps().get_double("MULTPV"),
+                                data::TargetType::INIT); 
+                }
+                writeSimulatorPropertiesLGRCell(grid, multipliers, initFile,global_fathers);
+            }
```

Comment:
Please create a helper function for this loop and call it from the body of the conditional instead of inserting the logic inline.